src/FSharp.Core/local.fs

```diff
@@ -1195,4 +1200,47 @@ module internal Seq =
                 while (e.MoveNext()) do res <- e.Current
                 ValueSome(res)
             else
-                ValueNone
\ No newline at end of file
+                ValueNone
+
+module internal Random =
+    open System
+
+    let private executeRandomizer (randomizer: unit -> float) =
+        let value = randomizer()
+        if value < 0.0 || value >= 1.0 then
+            let argName = nameof randomizer
+            invalidArgOutOfRangeFmt argName
+                "{0}\n{1} returned {2}, should be in range [0.0, 1.0)."
+                [|SR.GetString SR.outOfRange; argName; value|]
+        value
+
+    let next (randomizer: unit -> float) (minValue: int) (maxValue: int) =
+        maxValue - minValue
+        |> float
+        |> (*) (executeRandomizer randomizer)
+        |> int
+        |> (+) minValue
```

Comment:
Nit: I think this is easier to read when written parenthesized instead of pipelined

```suggestion
    let next (randomizer: unit -> float) (minValue: int) (maxValue: int) =
        int (executeRandomizer randomizer * float (maxValue - minValue)) + minValue
```

Reply:
Hm... I think one more set of parenthesis is needed, do you still think it's more readable? I actually wrote it like that initially, but then rewrote to pipes for readability :)
```fsharp
int ((executeRandomizer randomizer) * float (maxValue - minValue)) + minValue
```

Reply:
It is. Please don't pipe into partially applied operators, it's unreadable.

Reply:
> I think one more set of parenthesis is needed

You don't need parens around `executeRandomizer randomizer` (regular function application has higher precedence than infix operator application), unless you think it makes it clearer.

Reply:
Ok, changed it